source/common/utility.cc

```diff
@@ -62,4 +64,23 @@ Envoy::Network::DnsLookupFamily Utility::parseAddressFamilyOptionString(absl::st
   throw NighthawkException("Invalid argument");
 }
 
+void Utility::parseCommand(TCLAP::CmdLine& cmd, const int argc, const char* const* argv) {
+  cmd.setExceptionHandling(false);
+  try {
+    cmd.parse(argc, argv);
+  } catch (TCLAP::ArgException& e) {
```

Comment:
I don't really love how we're using exceptions so heavily for control-flow, which I guess means I don't really love TCLAP. I feel like it might be more Envoy-ish (particularly as a server, rather than a CLI tool) to just take a configuration file, similar to Envoy's bootstrap proto.

I guess I could say this is maybe a small argument in favor of making this service an Envoy filter, like I suggested above, but I'll definitely admit I don't know how much harder it'd be.

Reply:
Ultimately we want any non-trivial config for this to be proto-driven. As long as it's as simple as it is today, CLI is fine. The NH CLI tool needs to have a decent CLI interface to be competitive with other tools, but NH-as-a-service is only used in automation most likely.

Reply:
Ack